src/client/websocket/WebSocketShard.js

```diff
@@ -737,22 +823,30 @@ class WebSocketShard extends EventEmitter {
       this._emitDestroyed();
     }
 
-    // Step 2: Null the connection object
-    this.connection = null;
-
-    // Step 3: Set the shard status to DISCONNECTED
-    this.status = Status.DISCONNECTED;
-
-    // Step 4: Cache the old sequence (use to attempt a resume)
-    if (this.sequence !== -1) this.closeSequence = this.sequence;
+    /**
+     * Just to make sure the readyState is not stuck at CLOSING incase of a closeCode 4009.
+     * we can use this for other close codes as well but right now so far only 4009 is failing to close.
```

Comment:
```suggestion
    /*
     * Just to make sure the readyState is not stuck at CLOSING in case of a closeCode 4009.
     * We can use this for other close codes as well but so far only 4009 is failing to close.
```